tests/iothreads_test.go

```diff
@@ -0,0 +1,206 @@
+			{
+			    Name:       "ded1",
+			    VolumeName: "ded1volume",
+			    DiskDevice: v1.DiskDevice{
+				Disk: &v1.DiskTarget{
+				    Bus: "virtio",
+				},
+			    },
+			    DedicatedIOThread: &_true,
+			},
+			{
+			    Name:       "shr1",
+			    VolumeName: "shr1volume",
+			    DiskDevice: v1.DiskDevice{
+				Disk: &v1.DiskTarget{
+				    Bus: "virtio",
+				},
+			    },
+			},
+			{
+			    Name:       "shr2",
+			    VolumeName: "shr2volume",
+			    DiskDevice: v1.DiskDevice{
+				Disk: &v1.DiskTarget{
+				    Bus: "virtio",
+				},
+			    },
+			},
+		    },
+		},
+	    },
+	    Volumes: []v1.Volume{
+		{
+		    Name: "ded1volume",
+		    VolumeSource: v1.VolumeSource{
+			EmptyDisk: &v1.EmptyDiskSource{
+			    Capacity: resource.MustParse("1G"),
+			},
+		    },
+		},
+		{
+		    Name: "shr1volume",
+		    VolumeSource: v1.VolumeSource{
+			EmptyDisk: &v1.EmptyDiskSource{
+			    Capacity: resource.MustParse("1G"),
+			},
+		    },
+		},
+		{
+		    Name: "shr2volume",
+		    VolumeSource: v1.VolumeSource{
+			EmptyDisk: &v1.EmptyDiskSource{
+			    Capacity: resource.MustParse("1G"),
+			},
+		    },
+		},
+	    },
+	}
+
+	var IOThreadVMI *v1.VirtualMachineInstance
+
+	BeforeEach(func() {
+		tests.BeforeTestCleanup()
+		IOThreadVMI = tests.NewRandomVMI()
+		IOThreadVMI.Spec = IOThreadSpec
+	})
+
+	Context("IOThreads Policies", func() {
+
+		// CNV-1046
+		It("Virtual Disk Settings - IOThreads", func() {
+
+			// How many symbols in the names of pod and VMI shoul match
+			symbols_to_compare := 30
+
+			IOThreadVMI, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(IOThreadVMI)
+			// Step 1
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(IOThreadVMI.Name) > symbols_to_compare).To(BeTrue())
+
+			// get VMI from the pod
+			listOptions := metav1.ListOptions{}
+			podList, err := virtClient.CoreV1().Pods(tests.NamespaceTestDefault).List(listOptions)
+			// Step 2
+			Expect(err).ToNot(HaveOccurred())
+			Expect(podList.Items).To(HaveLen(1))
+			Expect(podList.Items[0].Name).To(HavePrefix("virt-launcher-" + IOThreadVMI.Name[:symbols_to_compare]))
+
+			getOptions := metav1.GetOptions{}
+			resultVMI, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(IOThreadVMI.Name, &getOptions)
+			// Step 3
+			Expect(err).ToNot(HaveOccurred())
+
+			// Step 4.1
+			Expect(*resultVMI.Spec.Domain.IOThreadsPolicy).To(Equal(sharedPolicy))
+			// Step 4.2
+			ded1_present := false
+			for _, disk := range resultVMI.Spec.Domain.Devices.Disks {
+				if disk.Name == "ded1" {
+					ded1_present = true
+					Expect(*disk.DedicatedIOThread).To(BeTrue())
+				}
+			}
+			Expect(ded1_present).To(BeTrue())
+
+			// Step 5 is NA for automatic testing
+
+			duration := time.Duration(60)*time.Second
+			time.Sleep(duration)
+			//tests.WaitUntilVMIReadyWithNamespace(tests.NamespaceTestDefault, resultVMI, tests.LoggedInCirrosExpecter)
+			command := "/usr/local/bin/oc project kubevirt-test-default && "
```

Comment:
In core, we have a helper function to accomplish this https://github.com/kubevirt/kubevirt/blob/6e6c1e42f2e78c3d6b0e8377a5550ada8ddb5a49/tests/utils.go#L1867

I believe you can build your own with this idea :).

Reply:
I seen it. I especially used another way to be sure that qemu run exactly what I check. Because It's test - I think bad idea use here implemented function.

Reply:
I see. If you are really writing such manual cases, it would be much easier to use different approach, than golang.

Reply:
Please use the method for getting the XML domain no need to rewrite code.